### DISCUSSION TITLE
Is there an outer margin in the giu?

### DISCUSSION THREAD

**Discussion by Seikaijyu** [AllenDang/giu#569] [Q&A]
Just now, I encountered a problem. The internal space of Child will be filled by the default internal margin. Is there an external margin in the giu? Or, can we solve this problem?

**Comment by Seikaijyu** [comment]
This will make it difficult for sub elements to calculate usable space

**Comment by gucio321** [comment]
There is a StyleVar for it (something with padding in it's name probably)

**Reply by Seikaijyu** [reply]
I tried StyleVarFramePadding, but it obviously modifies the internal components of Child

**Reply by Seikaijyu** [reply]
![image](https://user-images.githubusercontent.com/68497295/195645513-e4b20bab-4cd4-40a1-ba62-05abb9d3175e.png)

**Reply by gucio321** [reply]
🤔 maybe try to use imgui.pushstylevar in giu.custom widget to affect only the child widget...

**Reply by Seikaijyu** [reply]
Hmm... I'm going to try to customize the components now

**Reply by AllenDang** [reply]
Run imguidemo, open style editor, try style var like inner space or outter space.
